src/LottoMachineEntities/WinningCount.js

```diff
@@ -0,0 +1,47 @@
+import { LOTTO_WIN_RANK } from "../constants/rule.js";
+
+class WinningCount {
+  #winningCount;
+
+  constructor() {
+    this.#winningCount = Object.keys(LOTTO_WIN_RANK).reduce((acc, rank) => {
+      acc[rank] = 0;
+      return acc;
+    }, {});
+  }
+
+  static getRankType(matchCount, isBonusMatch) {
+    const [rankType] = Object.entries(LOTTO_WIN_RANK)
+      .filter(
+        ([, details]) =>
+          details.matchCount === matchCount && details.requiresBonus === isBonusMatch,
+      )
+      .map(([rank]) => rank);
+
+    return rankType;
+  }
+
+  increaseRankCount(rank) {
+    this.#winningCount[rank] += 1;
+  }
+
+  getWinningRankCountString() {
+    return `${LOTTO_WIN_RANK.threeMatch.string} (${LOTTO_WIN_RANK.threeMatch.prize.toLocaleString()}원) - ${this.#winningCount.threeMatch}개
+${LOTTO_WIN_RANK.fourMatch.string} (${LOTTO_WIN_RANK.fourMatch.prize.toLocaleString()}원) - ${this.#winningCount.fourMatch}개
+${LOTTO_WIN_RANK.fiveMatch.string} (${LOTTO_WIN_RANK.fiveMatch.prize.toLocaleString()}원) - ${this.#winningCount.fiveMatch}개
+${LOTTO_WIN_RANK.fiveMatchAndBonus.string} (${LOTTO_WIN_RANK.fiveMatchAndBonus.prize.toLocaleString()}원) - ${this.#winningCount.fiveMatchAndBonus}개
+${LOTTO_WIN_RANK.allMatch.string} (${LOTTO_WIN_RANK.allMatch.prize.toLocaleString()}원) - ${this.#winningCount.allMatch}개`;
+  }
+
+  calculateProfit() {
+    return (
+      this.#winningCount.threeMatch * LOTTO_WIN_RANK.threeMatch.prize +
+      this.#winningCount.fourMatch * LOTTO_WIN_RANK.fourMatch.prize +
+      this.#winningCount.fiveMatch * LOTTO_WIN_RANK.fiveMatch.prize +
+      this.#winningCount.fiveMatchAndBonus * LOTTO_WIN_RANK.fiveMatchAndBonus.prize +
+      this.#winningCount.allMatch * LOTTO_WIN_RANK.allMatch.prize
+    );
```

Comment:
현재 `winningCount`의 키와` LOTTO_WIN_RANK`의 키가 동일하니` forEach`나 `reduce`를 사용하면 더 간단히 나타낼 수 있다고 생각합니다!!

```suggestion
      let profit = 0;
      Object.entries(this.#winningCount).forEach(([rankType, count])=>{
           profit += count * LOTTO_WIN_RANK[rankType]
      })
      return profit
    );
```

Reply:
헉! 이부분은 놓쳤네요.. 발견해주셔서 감사합니다! 훨씬 가독성도 좋아지고, 코드도 불필요하게 길어지지 않네요 🤦🏻‍♀️ 
감사합니다! ☺️